Documentation/ExtensionArchitecture/Extbase/Reference/Validation/Validators/Index.rst

```diff
@@ -0,0 +1,624 @@
+=================
+
+Ensures a value is a valid \DateTimeInterface.
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'DateTime'])]
+    protected mixed $startDate;
+
+..  _extbase-validator-disjunction:
+
+DisjunctionValidator
+====================
+
+The `DisjunctionValidator` is a composite Extbase validator that allows you to
+combine multiple validators using a **logical OR**.
+
+It is the inverse of the `ConjunctionValidator`: the value is considered valid
+if **at least one** of the nested validators succeeds.
+
+**Behavior:**
+-   All validators are evaluated in order.
+-   Validation stops as soon as one validator passes.
+-   If **all validators fail**, their errors are merged and returned.
+-   If **any validator passes**, the result is considered valid.
+
+**Use cases:**
+
+Use this validator when a value is allowed to match **one of multiple conditions**.
+For example:
+-   A field can be either empty or a valid email
+-   A string can be either a number or "N/A"
+-   A value can match one of multiple formats
+
+**Usage:**
+
+This validator is typically used manually in custom validators or
+in validator resolver configurations.
+
+..  include:: _NoExamples.rst.txt
+
+..  note::
+
+    Extbase does **not** automatically construct disjunctions for you.
+    You must manually create and configure a `DisjunctionValidator` when needed.
+
+    It is not currently possible to use this validator directly via
+    `#[Validate(...)]` annotations.
+
+..  _extbase-validator-emailaddress:
+
+EmailAddressValidator
+=====================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\EmailAddressValidator`
+an email address using method :php:`\TYPO3\CMS\Core\Utility\GeneralUtility::validEmail()`.
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'EmailAddress'])]
+    protected string $email;
+
+..  _extbase-validator-file-name:
+
+FileNameValidator
+=================
+
+The :php-short:`TYPO3\CMS\Extbase\Validation\Validator\FileNameValidator`
+validates, that the given :php-short:`\TYPO3\CMS\Core\Http\UploadedFile`
+:php-short:`\TYPO3\CMS\Extbase\Persistence\ObjectStorage` of `UploadedFile`
+objects does not contain a PHP executable file by checking the given file
+extension.
+
+Internally the :php:`\TYPO3\CMS\Core\Resource\Security\FileNameValidator` is
+used to validate the file name.
+
+..  include:: _NoExamples.rst.txt
+
+..  _extbase-validator-file-size:
+
+FileSizeValidator
+=================
+
+The :php-short:`TYPO3\CMS\Extbase\Validation\Validator\FileSizeValidator`
+validates, that the given :php-short:`\TYPO3\CMS\Core\Http\UploadedFile`
+:php-short:`\TYPO3\CMS\Extbase\Persistence\ObjectStorage` of `UploadedFile`
+objects do not exceed the file size configured via the options.
+
+Options:
+
+`minimum`
+    The minimum file size to accept
+`maximum`
+    The maximum file size to accept
+
+Internally the :php:`\TYPO3\CMS\Core\Resource\Security\FileNameValidator` is
+used to validate the file name.
```

Comment:
This paragraph is misplaced and should be dropped.

Reply:
Done